### ISSUE TITLE
Using Assignment in AssignmentsStorageInterface::add()

### ISSUE THREAD

**Issue by githubjeka** [yiisoft/rbac#134]
An assigment has the `createdAt` attribute and you can't control it in AssignmentsStorageInterface::add() 

```php
public function add(string $itemName, string $userId): void;
```

Usually inside this a storage method the `time()` is called. 

For example, I found wrong logic with `createdAt` in `php-rbac` package and now hard to write a test for check it

How about rewrite this method 
```php
public function add(Assignment $assignment): void;
```

**Comment by samdark** [comment]
It was like that initially then we've simplified the API a bit removing DTOs.

**Comment by vjik** [comment]
Yes. This is need for fix https://github.com/yiisoft/rbac-php/issues/50.

Need fix `Manager::assign()` also. This can be several ways:

1) Add to `Manager::assign()` required parameter `createdAt`.
2) Add to `Manager::assign()` optional parameter `createdAt` with current time as default value.
3) Use current time into `Manager::assign()`.

**Comment by arogachev** [comment]
Closed by #190 and #170.
